subprojects/logging/src/integTest/groovy/org/gradle/internal/logging/console/taskgrouping/RichVerboseConsoleTypeFunctionalTest.groovy

```diff
@@ -44,22 +48,84 @@ class RichVerboseConsoleTypeFunctionalTest extends AbstractConsoleGroupedTaskFun
                 dependsOn helloWorld, byeWorld, silence
             }
         """
-    }
-
-    @Unroll
-    def "can have verbose task output according to --console"() {
         when:
-        executer.withConsole(mode)
         succeeds('all')
 
         then:
-        result.groupedOutput.task(':helloWorld').output == HELLO_WORLD_MESSAGE
-        result.groupedOutput.task(':byeWorld').output == BYE_WORLD_MESSAGE
+        result.groupedOutput.task(':helloWorld').output == 'Hello world'
+        result.groupedOutput.task(':byeWorld').output == 'Bye world'
         hasSilenceTaskOutput == result.groupedOutput.hasTask(':silence')
 
         where:
         mode    | hasSilenceTaskOutput
         Rich    | false
         Verbose | true
     }
+
+    def "long running task's output are divided by other output"() {
+        given:
+        buildFile << '''
+def sleepFor(String desc, int times){
+    (1..times).each {
+         Thread.sleep(1000)
+         println "${desc} wake up ${it}"
+    }
+}
```

Comment:
It looks like this test is correctly structured to avoid flaky scenario, however, I would try to align the implementation with [a similar test](https://github.com/gradle/gradle/blob/master/subprojects/logging/src/integTest/groovy/org/gradle/internal/logging/console/taskgrouping/BasicGroupedTaskLoggingFunctionalSpec.groovy#L123-L204) to make it easier to maintain.

Reply:
Good advice. I didn't notice that test and I think it's acceptable to "hijack" that test to test our new features so I slightly modified that test and remove this test to reduce maintaining burden.